src/Compilers/CSharp/Test/Emit2/Semantics/ExperimentalAttributeTests.cs

```diff
@@ -77,18 +80,779 @@ public class C
+public class C
+{
+    public virtual void M() { }
+}
+""";
+
+        var src = """
+public class Derived : C { public override void M() { } }
+""";
+
+        var comp = CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc });
         comp.VerifyDiagnostics();
     }
 
     [Fact]
-    public void OnModule()
+    public void OnOverride_ExperimentalFromAssembly_UsedFromMetadata()
     {
-        // Ignored on modules
         var libSrc = """
-[module: System.Diagnostics.CodeAnalysis.Experimental("DiagID1")]
+[assembly: System.Diagnostics.CodeAnalysis.Experimental("Diag")]
 public class C
+{
+    public virtual void M() { }
+}
+""";
+
+        var src = """
+public class Derived : C { public override void M() { } }
+""";
+
+        var comp = CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        // CONSIDER narrowing the location on constructor initializer obsolete/experimental attributes
+        comp.VerifyDiagnostics(
+            // 0.cs(1,1): warning Diag: 'C.C()' is for evaluation purposes only and is subject to change or removal in future updates.
+            // public class Derived : C { public override void M() { } }
+            Diagnostic("Diag", "public class Derived : C { public override void M() { } }").WithArguments("C.C()").WithLocation(1, 1),
+            // 0.cs(1,24): warning Diag: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
+            // public class Derived : C { public override void M() { } }
+            Diagnostic("Diag", "C").WithArguments("C").WithLocation(1, 24)
+            );
+    }
+
+    [Theory, CombinatorialData]
+    public void OnExplicitMethodImplementation(bool inSource)
+    {
+        var libSrc = """
+public interface I
+{
+    [System.Diagnostics.CodeAnalysis.Experimental("Diag")]
+    public void M();
+}
+""";
+
+        var src = """
+public class C : I
+{
+    void I.M() { }
+}
+""";
+
+        var comp = inSource
+         ? CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc })
+         : CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        comp.VerifyDiagnostics();
+    }
+
+    [Theory, CombinatorialData]
+    public void OnExplicitMethodImplementation_Obsolete(bool inSource)
+    {
+        var libSrc = """
+public interface I
+{
+    [System.Obsolete("message", true)]
+    public void M();
+}
+""";
+
+        var src = """
+public class C : I
+{
+    void I.M() { }
+}
+""";
+
+        var comp = inSource
+         ? CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc })
+         : CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        comp.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void MissingAssemblyAndModule()
```

Comment:
>MissingAssemblyAndModule [](http://example.com/codeflow?start=16&length=24)

Is this testing `[Experimental]` support?

Reply:
Same answer as for VB